### DISCUSSION TITLE
Use Read Aloud in PDF with Thorium Reader

### DISCUSSION THREAD

**Discussion by togregg** [edrlab/thorium-reader#1844] [Q&A]
Is there any thought of allowing PDFs to use the Read Aloud feature in Thorium Reader in the future? Thorium can read EPUBS and DAISY files, but I couldn't find anything in the documentation if reading PDFs is supported or not.

Sincerely,

Tomas Gregg

**Comment by danielweck** [comment]
Hello! Thorium's Text-To-Speech experience relies on the structure and semantics of authored HTML documents (which can themselves contain foreign markup such as SVG and MathML).
Interestingly, under the hood PDF rendering in Thorium is performed by a Javascript library called PDF.js, which generates its own custom DOM and CSS styling rules. So in principle we could potentially hook into PDF.js' rendered markup and generate the TTS audio stream + navigation, as we do for regular HTML documents. Unfortunately in practice this is likely to result in gibberish, because the markup generated by PDF.js is constructed mainly to visually mirror the consumed PDF document. In other words, the aim is pixel-perfect rendition, not correct flow of text / reading order (which most PDF files lack in the first place, anyway).
